### DISCUSSION TITLE
Remove key or mouse binding action during the game?

### DISCUSSION THREAD

**Discussion by chengenzhao** [AlmasB/FXGL#1144] [Q&A]
Hi:

Is there any way to remove specific key action or mouse action during the game?
I tried these code and doesn't work
```java
    var action = FXGL.getInput().getActionByName(KeyCode.ENTER.name());
    var result = FXGL.getInput().getAllBindings().remove(action);

    FXGL.getInput().addAction(new UserAction(KeyCode.ENTER.name()) {
      @Override
      protected void onActionBegin() {
        System.out.println("rebind failed");
      }

      @Override
      protected void onActionEnd() {

      }
    }, KeyCode.ENTER);

```

Thanks

**Comment by AlmasB** [comment]
As far as I can remember, no I don't think there is any API that allows removing, only rebinding. I think the idea was that all actions should be pre-defined before the game starts and remain active for the entire game.

If there are any use cases you can suggest, then we should be able to turn this into an issue and fix it.

**Reply by chengenzhao** [reply]
For example, we have a GI Joe who runs into battlefield
and there are some saving points in the filed
when he enters these areas(addCollisionHandler), the prompt message will jump out like (press S to save the game or open the saving menu)
and if he exits this area, of course pressing S won't work
If you could add actions and remove as we want
we could add pressing S action when he enters the saving areas and remove them when he exits the area

**Reply by chengenzhao** [reply]
And for rebind method
If there exists an action then rebind will fail
rebind only works for the key code without action and if so, why wouldn't use addAction directly?
Is it original design like this?
I think the correct way is allow user to change the action behavior using rebind method, right?
